### PR TITLE
FEATURE: EGL-247 - allow to show unrelated date items on KD

### PR DESCRIPTION
**Pull request by vannguyengd**
feat: allow to show unrelated date items on KD
JIRA: EGL-247

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16074/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17559/ : FAILURE in 8m 41s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16077/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17564/ : SUCCESS in 9m 18s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16080/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17567/ : SUCCESS in 10m 16s

File: libs/sdk-ui-dashboard/src/presentation/widget/common/configuration/utils.ts [hunk 1]
```diff
@@ -47,3 +53,72 @@ export function removeDateFromTitle(title: string): string {
...
+}
+
+export function getDateConfigurationDropdownHeight(
+    dropdownButtonTop: number,
+    dropdownButtonHeight: number,
+    dropdownBodyHeight: number,
+    hasFooter = false,
+) {
+    const nodeAboveHeight =
+        dropdownButtonTop - 2 * DATE_DROPDOWN_BODY_MARGIN - PADDING - Number(hasFooter) * UNRELATED_HEIGHT;
```
[the remark this answers is not in the document]
Comment (pbenes) [reply libs/sdk-ui-dashboard/src/presentation/widget/common/configuration/utils.ts:79]: Or maybe even better, separating the whole fixed size to the constant and use 
```
const footerHeight = hasFooter ? UNRELATED_HEIGHT : 0;
const height = 2 * DATE_DROPDOWN_BODY_MARGIN + PADDING + footerHeight
const nodeAboveHeight = dropdownButtonTop - height;
...
...
        window.innerHeight -
        (dropdownButtonTop +
            dropdownButtonHeight +
            height);
```

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16089/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17577/ : SUCCESS in 9m 11s

File: libs/sdk-ui-dashboard/src/model/store/config/configSelectors.ts [hunk 2]
```diff
@@ -563,3 +563,15 @@ export const selectIsDrillDownEnabled: DashboardSelector<boolean> = createSelect
         return isKPIDashboardImplicitDrillDownEnabled || isAttribueHierarchiesEnabled;
     },
 );
+
+/**
+ * Returns whether attribute hierarchies are enabled.
+ *
+ * @internal
+ */
+export const selectEnableUnavailableItemsVisibility: DashboardSelector<boolean> = createSelector(
```
[the remark this answers is not in the document]
Comment (vannguyengd) [reply libs/sdk-ui-dashboard/src/model/store/config/configSelectors.ts:572]: I have updated this. Thank you!

File: libs/sdk-ui-dashboard/src/presentation/widget/common/configuration/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vannguyengd) [reply libs/sdk-ui-dashboard/src/presentation/widget/common/configuration/utils.ts]: I copied this function from AD, thank you for suggesting to make it more readable!

File: libs/sdk-ui-dashboard/src/presentation/widget/common/configuration/utils.ts [hunk 3]
```diff
@@ -47,3 +53,72 @@ export function removeDateFromTitle(title: string): string {
...
+    shouldDisplayUnrelatedDatasets = false,
+): (IDateDataset | IDateDatasetHeader)[] {
+    const sortedUnrelatedDateDataset = sortDateDatasets(
+        (unrelatedDateDatasets || []).map(catalogDateDatasetToDateDataset),
+    );
+    const withUnrelatedHeader = [unrelatedHeader, ...sortedUnrelatedDateDataset];
+
+    if (shouldDisplayUnrelatedDatasets) {
+        return [
+            ...sortDateDatasets(relatedDateDatasets.map(catalogDateDatasetToDateDataset)),
```
[the remark this answers is not in the document]
Comment (vannguyengd) [reply libs/sdk-ui-dashboard/src/presentation/widget/common/configuration/utils.ts]: It also needs to be displayed together with unrelated datasets. I have updated this. Thank you!

**Review by pbenes — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4174

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3287/ : SUCCESS in 10m 54s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17580/ : SUCCESS in 9m 17s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/900/ : SUCCESS in 9m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4447/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1167/ : FAILURE in 1m 21s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/805/ : SUCCESS in 8m 24s

**Pull request merged**